internal/providers/pluginfw/resources/notificationdestinations/data_notification_destinations.go

```diff
@@ -0,0 +1,126 @@
+}
+
+func (d *NotificationDestinationsDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = "databricks_notification_destinations_pluginframework"
+}
+
+func (d *NotificationDestinationsDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: tfschema.DataSourceStructToSchemaMap(NotificationDestinationsInfo{}, nil),
+	}
+}
+
+func (d *NotificationDestinationsDataSource) Configure(_ context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	if d.Client == nil {
+		d.Client = pluginfwcommon.ConfigureDataSource(req, resp)
+	}
+}
+
+func validateType(notificationType string) diag.Diagnostics {
+	validTypes := map[string]struct{}{
+		string(settings.DestinationTypeEmail):          {},
+		string(settings.DestinationTypeMicrosoftTeams): {},
+		string(settings.DestinationTypePagerduty):      {},
+		string(settings.DestinationTypeSlack):          {},
+		string(settings.DestinationTypeWebhook):        {},
+	}
+
+	if _, valid := validTypes[notificationType]; !valid {
+		return diag.Diagnostics{diag.NewErrorDiagnostic(fmt.Sprintf("Invalid type '%s'; valid types are EMAIL, MICROSOFT_TEAMS, PAGERDUTY, SLACK, WEBHOOK.", notificationType), "")}
+	}
+	return nil
+}
+
+func validateLength(destinations []settings_tf.ListNotificationDestinationsResult) diag.Diagnostics {
+	if len(destinations) == 0 {
+		return diag.Diagnostics{diag.NewErrorDiagnostic("Could not find any notification destinations with the specified criteria.", "")}
+	}
+	return nil
+}
+
+func appendToResponse(resp *datasource.ReadResponse, diags diag.Diagnostics) bool {
+	resp.Diagnostics.Append(diags...)
+	return resp.Diagnostics.HasError()
+}
+
+func (d *NotificationDestinationsDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	w, diags := d.Client.GetWorkspaceClient()
+	if appendToResponse(resp, diags) {
+		return
+	}
+
+	var notificationInfo NotificationDestinationsInfo
+	if appendToResponse(resp, req.Config.Get(ctx, &notificationInfo)) {
+		return
+	}
+
+	notificationType := notificationInfo.Type.ValueString()
+	notificationDisplayName := notificationInfo.DisplayNameContains.ValueString()
+
+	if notificationType != "" {
+		if appendToResponse(resp, validateType(notificationType)) {
+			return
+		}
+	}
+
+	notificationsGoSdk, err := w.NotificationDestinations.ListAll(ctx, settings.ListNotificationDestinationsRequest{})
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to fetch Notification Destinations", err.Error())
+		return
+	}
+
+	var notificationsTfSdk []settings_tf.ListNotificationDestinationsResult
+	for _, notification := range notificationsGoSdk {
+		if (notification.DestinationType.String() != notificationType) || (notificationDisplayName != "" && !strings.Contains(notification.DisplayName, notificationDisplayName)) {
```

Comment:
We need to discuss if it makes sense to have case-insensitive comparison of display name

Reply:
@alexott Agreed, makes sense. Since we're not sending the `display_name` down as a filter to the call but applying it at the client-side, we could just make sure both values are wrapped with `strings.ToLower()`  and then do the comparison.  LMK what you think.